### PR TITLE
Add ClearHistory & DeactivateChat methods

### PR DESCRIPTION
**Pull request by cammellos**
Adds 2 methods, clear history and deactivate chat.
These are functionalities that are already used in the client, but done through a mixture of different endpoints.
This commit consolidates the logic so it can be done on a single call.
It also adds a few endpoints:
- AddContact
- RemoveContact

But those are still wip.
It moves also a bunch of messenger methods on their own file for clarity.

### CODE REVIEW

**Comment by jakubgs** [comment]
Why is this adding `AddContact` and `RemoveContact` if they are still work in progress?
Could we at least do it as a separate commit or are the changes connected to the other changes?

File: protocol/contact.go [hunk 1]
```diff
@@ -83,6 +82,16 @@ func (c Contact) IsBlocked() bool {
...
 }
 
+func (c *Contact) Remove() {
+	var newSystemTags []string
+	for _, tag := range newSystemTags {
+		if tag != contactAdded {
+			newSystemTags = append(newSystemTags, tag)
+		}
+	}
+	c.SystemTags = newSystemTags
```
Comment (jakubgs) [inline protocol/contact.go:94]: I don't get what this method is doing.

File: protocol/message_persistence.go [hunk 2]
```diff
@@ -1131,3 +1150,78 @@ func (db sqlitePersistence) InvitationByID(id string) (*GroupChatInvitation, err
...
+	err := db.saveChat(tx, *chat)
+	if err != nil {
+		return err
+	}
+
+	return db.clearHistory(chat, currentClockValue, tx)
+}
+
+func (db sqlitePersistence) clearHistory(chat *Chat, currentClockValue uint64, tx *sql.Tx) error {
+	// Set deleted at clock value
```
Comment (jakubgs) [inline protocol/message_persistence.go]: That's not a very helpful comment, we can see in this line that's what happens:
```go
chat.DeletedAtClockValue = currentClockValue
```
But why does it need to be set? That's what the comment should say.

File: protocol/messenger.go [hunk 3]
```diff
@@ -2089,59 +2089,40 @@ func (m *Messenger) DeleteChat(chatID string) error {
...
+		return nil, err
 	}
 
-	contact.Identicon = identicon
-	contact.Alias = name
-
-	if m.isNewContact(contact) || m.hasNicknameChanged(contact) {
-		err := m.syncContact(context.Background(), contact)
+	if chat.Public() {
+		err := m.reregisterForPushNotifications()
```
Comment (jakubgs) [inline protocol/messenger.go:2109]: "reregister"? Shouldn't this be called "deregister"? Or am I missing what this actually does?

File: protocol/messenger.go [hunk 4]
```diff
@@ -2089,59 +2089,40 @@ func (m *Messenger) DeleteChat(chatID string) error {
...
-
-	m.allContacts[contact.ID] = contact
+	m.allChats[chatID] = chat
 
-	// Reregister only when data has changed
-	if shouldReregisterForPushNotifications {
-		return m.reregisterForPushNotifications()
-	}
+	response.Chats = []*Chat{chat}
+	// TODO: Remove filters
```
Comment (jakubgs) [inline protocol/messenger.go:2118]: Is this TODO important?

**Review by jakubgs — APPROVED** [review]
I don't see anything major wrong with this. Just some tiny nitpicks.

File: protocol/contact.go (hunk 1 above)
Comment (cammellos) [reply protocol/contact.go:94]: It's removing the `newSystemTag` from a contact (effectively removing the contact). I will add a comment

File: protocol/messenger.go (hunk 3 above)
Comment (cammellos) [reply protocol/messenger.go:2109]: it's re-registering effectively, as the fact that a contact has been removed means that we need to re-register with the PN server as our options have changed, I can add a comment

**Pull request merged**